src/Compilers/CSharp/Test/Semantic/Semantics/ExtensionTypeTests.cs

```diff
@@ -8158,4 +8232,1151 @@ public void ObsoleteOnType()
         // PROTOTYPE revisit once attributes are allowed on extension types.
         // The Obsolete poison attribute should not be emitted when the user marked the type as obsolete.
     }
+
+    [ConditionalFact(typeof(CoreClrOnly))]
+    public void MemberLookup()
+    {
+        var src = """
+E.M();
+_ = E.Property;
+
+public explicit extension E for object
+{
+    public static void M()
+    {
+        System.Console.Write("Method ");
```

Comment:
>System.Console.Write("Method "); [](http://example.com/codeflow?start=8&length=32)

I am curious, when we are inside the extension type, are we supposed to find members of the underlying type as well? Are we testing lookup inside an extension type?

Reply:
That scenario is covered by other sections of the spec:
1. "simple name" rules which are summarized as "After doing an unsuccessful member lookup in an extension, we'll also perform a member lookup in the underlying type."
2. "member access" rules which are summarized as "After doing an unsuccessful member lookup in a type, we'll perform an member lookup in the underlying type if we were dealing with an extension, and if that is still unsuccessful, we'll perform an extension member lookup."

Those scenarios will be covered in a later PR.